src/Sulu/Bundle/AdminBundle/Resources/js/containers/List/tests/adapters/ColumnListAdapter.test.js

```diff
@@ -111,6 +111,61 @@ test('Render different kind of data with edit button', () => {
     expect(columnListAdapter).toMatchSnapshot();
 });
 
+test('Render correct icon in edit button based on permissions', () => {
+    const data = [
+        [
+            {
+                // button should not be visible because view permission is missing
+                id: 1,
+                title: 'Missing view permission',
+                hasChildren: false,
+                _permissions: {
+                    view: false,
+                },
+            },
+            {
+                // button should contain eye icon because edit permission is missing
+                id: 1,
+                title: 'Missing edit permission',
+                hasChildren: false,
+                _permissions: {
+                    edit: false,
+                    view: true,
+                },
+            },
+            {
+                // button should contain pen icon because user has sufficient permissions
+                id: 1,
+                title: 'Sufficient Permissions',
+                hasChildren: false,
+                _permissions: {
+                    edit: true,
+                    view: true,
+                },
+            },
+            {
+                // button should contain plus icon because item is a ghost page
+                id: 1,
+                title: 'Ghost Page',
+                hasChildren: false,
+                ghostLocale: 'en',
+            },
+        ],
+        [],
+    ];
+
+    const columnListAdapter = render(
+        <ColumnListAdapter
+            {...listAdapterDefaultProps}
+            activeItems={[]}
+            data={data}
+            onItemClick={jest.fn()}
+        />
+    );
+
+    expect(columnListAdapter).toMatchSnapshot();
```

Comment:
I'd like to reduce the number of snapshot tests, since we've got way too many :see_no_evil: If you change something in a dumb component, that causes too many tests to fail, and people (including me) are too lazy to look at the snapshots in detail... I would suggest including these examples in the previous tests, which also includes a snapshot, or use clearer assertions here (then we also know what exactly the test is looking at).